lib/main_common.pm

```diff
@@ -1567,7 +1603,21 @@ sub load_extra_tests {
         load_extra_tests_desktop;
     }
     else {
-        load_extra_tests_textmode;
+        # Extra tests are too long, split the test into subtest according to the
+        # EXTRATEST variable; to maintain compatibility, run all tests if the
+        # variable is equal 1
+        if (check_var('EXTRATEST', 1)) {
+            load_extra_tests_textmode;
```

Comment:
that is a good and safe approach which I appreciate but in the current approach this causes too much code duplication, e.g. what is now the "other" tests that are duplicated in call. Better to call the specific load functions within the generic `load_extra_tests_textmode` to not duplicate too much.

Reply:
Yes, there is a lot of duplication (and this is also the source of the second, refactoring, commit). My idea is, after the merge, when the test settings in openQA is adjusted, to clean-up this whole mess and get rid of the old scenario. The idea behind this is to separate some tests from the generic ``load_extra_tests_textmode`` to be able to run independently. Unfortunately, those tests are not all on neighboring lines and I wanted to avoid change in ordering (not sure, if that can break something).

So, this should mean something like run the old code until we get rid of the ``EXTRATEST=1`` in all test suites.

Reply:
Hm, I understand but you know how reality goes, it might take ages to get rid of "EXTRATEST=1" and so we keep this duplication in place for long :/

If you want to change the order in the new approach anyway then you need to test if the order works so why not change the order already in the way as it would look like afterwards as a first change, have that merged, then your changes with loading helper functions and then finally get rid of the old schedule?

Reply:
Good point, @okurz, rewritten.